clientconn.go

```diff
@@ -361,11 +361,11 @@ func (cc *ClientConn) lbWatcher() {
 
 func (cc *ClientConn) newAddrConn(addr Address, skipWait bool) error {
 	ac := &addrConn{
-		cc:           cc,
-		addr:         addr,
-		dopts:        cc.dopts,
-		shutdownChan: make(chan struct{}),
+		cc:    cc,
+		addr:  addr,
+		dopts: cc.dopts,
 	}
+	ac.dopts.copts.Cancel = make(chan struct{})
```

Comment:
That sounds like a reasonable philosophy, but having more copies of random channels certainly does not make this code easier to read.

Perhaps you could refactor this as you see fit in a follow-up?